src/openfermion/measurements/_bayesian_estimators.py

```diff
@@ -0,0 +1,845 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""
+Classes for Bayesian estimators for quantum phase estimation.
+Estimators described in arXiv:1809.09697 (in particular in App.C and App.C.1)
+"""
+import warnings
+import numpy
+from numpy import cos, sin, pi
+from scipy import sparse, optimize
+from openfermion.config import EQ_TOLERANCE
+
+
+class FourierProbabilityDist(object):
+    """
+    Stores a multivariant Fourier representation of a periodic function:
+
+    f(phi_0,phi_1,...) = 
+        sum_j A_j sum_n (c_{j,2n} cos(n*phi_j) +
+                         c_{j,2n-1} sin(n*phi_j))
+
+    In particular, this class stores values of A_j, c_{j,2n},
+    c_{j,2n+1} from the above equation.
+
+    It also provides routines for multiplying f by functions of the form
+    cos^2(n*phi_j + beta), and for calculating maximum-likelihood
+    distributions of the A_j variables.
+
+    More details can be found in arXiv:1809.09697, Appendix C.
+    """
+
+    def __init__(self,
+                 amplitude_guess=None,
+                 amplitude_vars=None,
+                 num_vectors=1,
+                 num_freqs=1000,
+                 max_n=1,
+                 vector_guess=None):
+        """
+        Args:
+            amplitude_guess (numpy array or list):
+                estimates for the amplitude of the jth component of
+                the estimated function.
+            amplitude_vars (numpy array or list): variance on the
+                guess in amplitude_guess.
+            num_vectors (int): number of phases phi_j to estimate
+            num_freqs (int): number of wave components cos(n*phi_j)
+                to store coefficients for (this provides a cut-off to
+                the sensitivity of the function).
+            max_n (int): the maximum n to be used when multiplying
+                the stored function by matrices of the form
+                cos(n*phi_j + beta).
```

Comment:
This is unclear. What is "the maximum n"?

Reply:
So here I mean 'assuming the user wants to multiply their distributions by cos(n*phi_j + beta), what is the maximum n that they require'. I guess 'the maximum n to be used when multiplying...' -> 'the maximum n required for multiplying...'